File: 5eungwon/6/10828.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;
int main() {
	int t;
	int i;

	cin >> t;
	stack <int> good;
	string ccc = "";
	for (i = 0; i < t + 1; i++) {
		getline(cin, ccc);
		if (ccc == "pop") {
			if (good.size() == 0) {
				cout << -1 << endl;
				continue;
			}
			cout << good.top() << endl;
			good.pop();
		}
		else if (ccc == "size") {
			cout << good.size() << endl;
		}
		else if (ccc == "empty") {
			if (good.size() == 0) {
				cout << 1 << endl;
			}
			else if (good.size() != 0) {
				cout << 0 << endl;
			}
		}
		else if (ccc == "top") {
			if (good.size() == 0) {
				cout << -1 << endl;
				continue;
			}
			cout << good.top() << endl;
		}
		else if (ccc.substr(0, 4) == "push") {
			good.push(stoi(ccc.substr(5, ccc.length() - 5)));
		}
	}

}

File: topwon1/6/10828.py
import sys

stack = []
for i in range(int(sys.stdin.readline())):
    command = list(map(str,sys.stdin.readline().split()))
    if command[0] == "push":
        stack.append(command[1])
    elif command[0] == "pop":
        if len(stack) == 0:
            print(-1)
            continue
        print(stack.pop())
    elif command[0] == "size":
        print(len(stack))
    elif command[0] == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command[0] == "top":
        if len(stack) == 0:
            print(-1)
            continue
        print(stack[-1])